### PR TITLE
Refactor logging  +semver:minor

### PR DESCRIPTION
**Pull request by mholo65**
- Expose ILoggingBuilder instead of accepting a ILoggerFactory

Some open questions. How to maintain backwards compatibility, or should we?

### CODE REVIEW

**Comment by mholo65** [comment]
@david-driscoll is it worth it to work around and try to keep the old API with `ILoggingFactory` in order to don't break any existing users? We're still `0.x` so might as well break 😄 Need to fixup the tests however, and probably fix the minimum loglevel also once for all 🛩

File: test/Lsp.Tests/LanguageServerTests.cs [hunk 1]
```diff
@@ -37,10 +37,10 @@ public async Task Works_With_IWorkspaceSymbolsHandler()
...
                 //.WithHandler(handler)
                 .WithInput(process.ClientOutputStream)
-                .WithOutput(process.ClientInputStream)
-                .WithLoggerFactory(LoggerFactory)
-                .AddDefaultLoggingProvider()
-                .WithMinimumLogLevel(LogLevel.Trace),
+                .WithOutput(process.ClientInputStream),
+                //.WithLoggerFactory(LoggerFactory)
+                //.AddDefaultLoggingProvider()
+                //.WithMinimumLogLevel(LogLevel.Trace),
```
Comment (david-driscoll) [inline test/Lsp.Tests/LanguageServerTests.cs]: Can probably safely delete these lines

File: test/Lsp.Tests/LanguageServerTests.cs [hunk 2]
```diff
@@ -63,9 +63,9 @@ public async Task GH141_CrashesWithEmptyInitializeParams()
             var server = LanguageServer.PreInit(x => x
                 .WithInput(process.ClientOutputStream)
                 .WithOutput(process.ClientInputStream)
-                .WithLoggerFactory(LoggerFactory)
-                .AddDefaultLoggingProvider()
-                .WithMinimumLogLevel(LogLevel.Trace)
+                //.WithLoggerFactory(LoggerFactory)
```
Comment (david-driscoll) [inline test/Lsp.Tests/LanguageServerTests.cs]: Can probably safely delete these lines
